### DISCUSSION TITLE
Can't block totally LAN access

### DISCUSSION THREAD

**Discussion by peppelongo96** [pivpn/pivpn#1708] [Support]
Hi guys!
I need some help to configure my iptable to block every (type of) connection to my LAN IPs from outer
I followed this guide https://github.com/pivpn/pivpn/issues/1244#issuecomment-775077139 but I only managed to block the access to my gateway IP 192.168.1.254.
For example, I'm still able to perform every SSH connection and also to connect to a local web server deployed in a specific port.
The instruction added is:

`iptables -I FORWARD -i wg0 -o eth0 -s 10.218.198.3 -d 192.168.1.0/24 -j REJECT`

and the current ipatables configuration is:

```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
REJECT     all  --  10.218.198.3         192.168.1.0/24       reject-with icmp-port-unreachable
DOCKER-USER  all  --  anywhere             anywhere
DOCKER-ISOLATION-STAGE-1  all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain DOCKER (2 references)
target     prot opt source               destination
ACCEPT     tcp  --  anywhere             172.18.0.2           tcp dpt:5000

Chain DOCKER-ISOLATION-STAGE-1 (1 references)
target     prot opt source               destination
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere
RETURN     all  --  anywhere             anywhere

Chain DOCKER-ISOLATION-STAGE-2 (2 references)
target     prot opt source               destination
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
RETURN     all  --  anywhere             anywhere

Chain DOCKER-USER (1 references)
target     prot opt source               destination
RETURN     all  --  anywhere             anywhere
```
Could anyone help me? Thanks

**Comment by orazioedoardo** [comment]
> For example, I'm still able to perform every SSH connection and also to connect to a local web server deployed in a specific port.

Local web server on the PiVPN host?

**Reply by peppelongo96** [reply]
Yes, correct

**Reply by orazioedoardo** [reply]
Then you need to add the same rule but to the INPUT chain instead of the FORWARD chain.

Edit: wait, this will also kill your WireGuard connection.

Edit 2: allow from LAN and deny from VPN. Add in order:

```
iptables -I INPUT 1 -i eth0 -j ACCEPT
iptables -I INPUT 2 -i wg0 -j REJECT
```

**Reply by peppelongo96** [reply]
Thanks for the proper command too. What 1 and 2 is referring to?

**Reply by orazioedoardo** [reply]
Order of add, 1 means “add as first rule”. Order matters when traversing the firewall. If say rule N instructs the firewall to block a packet, and N+1 allows the packet, in the end the packet is blocked.

**Reply by peppelongo96** [reply]
okay thanks for the explanation. I'll implement all tonight :)

**Reply by peppelongo96** [reply]
> Then you need to add the same rule but to the INPUT chain instead of the FORWARD chain.
> 
> Edit: wait, this will also kill your WireGuard connection.
> 
> Edit 2: allow from LAN and deny from VPN. Add in order:
> 
> ```
> iptables -I INPUT 1 -i eth0 -j ACCEPT
> iptables -I INPUT 2 -i wg0 -j REJECT
> ```

Okay it is not working as what i need.
First, in this way any client from wireguard vpn can't access pivpn host. But i need that only specific clients must be refused.
Also, that configuration is rejecting SSH connections to pivpn host, but i'm still able to access the local webserver on the same host. To be more clear, pivpn host has 192.168.1.106, while local webserver is deplyed at 192.168.1.106:5000.
Thanks for your support

**Reply by orazioedoardo** [reply]
> But i need that only specific clients must be refused.

Ok, more on this later.

> Also, that configuration is rejecting SSH connections to pivpn host, but i'm still able to access the local webserver on the same host.

That's weird the second rule rejects everything from the WireGuard interface. What does you iptables currently look like?
